### PR TITLE
Saad/lp 1506

### PR DESCRIPTION
**Pull request by saad-waqar**
Added multiple rerun creation api

### CODE REVIEW

File: openedx/features/cms/urls.py [hunk 1]
```diff
@@ -2,5 +2,5 @@
 from . import views
 
 urlpatterns = [
-    url(r"^automate-rerun/$", views.automate_rerun, name="automate_rerun"),
+    url(r"^course_multiple_rerun/$", views.course_multiple_rerun_handler, name="course_multiple_rerun   "),
```
Comment (saad-waqar) [inline openedx/features/cms/urls.py]: space in url name

File: openedx/features/cms/helpers.py [hunk 2]
```diff
@@ -3,13 +3,16 @@
 
 from dateutil.parser import parse
 from courseware.courses import get_course_by_id
+from opaque_keys.edx.keys import CourseKey
```
Comment (emraan004) [inline openedx/features/cms/helpers.py]: Let's create a new app inside feature/cms directory and move all this code there

File: openedx/features/cms/helpers.py [hunk 3]
```diff
@@ -230,3 +233,96 @@ def calculate_date_by_delta(date, source_date, destination_date):
...
     date_delta = source_date - date
     return destination_date - date_delta
+
+
+def update_course_re_run_details(course_re_run_details):
+    """
+    This method gets new rerun data in dict and return it with necessary data for rerun creation
+    along with updated run id's for each rerun
+    :param course_re_run_details: dict containing rerun data for all courses
+    :return: dict with new run ids
```
Comment (emraan004) [inline openedx/features/cms/helpers.py]: Please make this comment descriptive

File: openedx/features/cms/helpers.py [hunk 4]
```diff
@@ -230,3 +233,96 @@ def calculate_date_by_delta(date, source_date, destination_date):
...
+    """
+    for course_detail in course_re_run_details:
+        course_key = CourseKey.from_string(course_detail['source_course_key'])
+        # replacing string course key with course key object in course rerun details dict
+        course_detail['source_course_key'] = course_key
+        source_course = modulestore().get_course(course_key)
+
+        if not source_course.end:
+            course_detail['error'] = 'Course ID not found'
```
Comment (emraan004) [inline openedx/features/cms/helpers.py]: let's make this msg is correct

File: openedx/features/cms/helpers.py [hunk 5]
```diff
@@ -230,3 +233,96 @@ def calculate_date_by_delta(date, source_date, destination_date):
...
+            course_detail['has_errors'] = True
+            break
+
+        run_number = calculate_next_rerun_number(source_course.id)
+        course_detail['display_name'] = source_course.display_name
+        course_detail['number'] = source_course.number
+        course_detail['org'] = source_course.org
+
+        runs = course_detail['runs']
+        for n, run in enumerate(runs):
```
Comment (saad-waqar) [inline openedx/features/cms/helpers.py]: `n` ==> some meaningful

File: openedx/features/cms/helpers.py [hunk 6]
```diff
@@ -230,3 +233,96 @@ def calculate_date_by_delta(date, source_date, destination_date):
...
+
+    new_run_id = "{}_{}_{}_{}".format(
+        run_number, run_dict['release_number'], run_dict['start'].strftime(RUN_DATE_FORMAT),
+        course_end_date.strftime(RUN_DATE_FORMAT)
+    )
+    return new_run_id
+
+
+def get_course_group(course_key):
+    course_rerun = CourseRerunState.objects.filter(course_key=course_key).first()
```
Comment (saad-waqar) [inline openedx/features/cms/helpers.py]: add doc strings

File: openedx/features/cms/helpers.py (hunk 2 above)
Comment (saad-waqar) [reply openedx/features/cms/helpers.py]: This change will be done in the consequent PR for multiple reruns

Comment (emraan004) [reply openedx/features/cms/helpers.py]: ok

**Pull request merged**